### ISSUE TITLE
Fix FP S4123 (`no-invalid-await`): don't flag when function returns a thenable object

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#4113]
When using `await foo()` and `foo()` returns a thenable object, a FP issue is raised by rule S4123.

Reproducer:

```ts
class Foo {
  then: Promise<Bar>;
}
function qux(): Foo {}
const baz = await qux();
```

Issue reported on community ticket: https://community.sonarsource.com/t/potential-false-positive-typescript-rule-s4123-with-mongoose-6-x/97482
